code/ModelAdmin.php

```diff
@@ -183,6 +185,65 @@ public function Link($action = null)
         return parent::Link($action);
     }
 
+    /**
+     * Get the link for the tab of a DataObject subclass managed by this ModelAdmin.
+     *
+     * Note that if the modelClass is managed on multiple tabs and you want a link for a
+     * specific tab you should use {@link getLinkForModelTab} instead.
+     *
+     * @throws InvalidArgumentException if $modelClass is not managed by this ModelAdmin.
+     */
+    public function getLinkForModelClass(string $modelClass): string
+    {
+        if (!$this->isManagedModel($modelClass)) {
+            throw new InvalidArgumentException("$modelClass isn't managed by this ModelAdmin.");
+        }
+        return $this->getLinkForModelTab($this->getModelTabForModelClass($modelClass));
+    }
+
+    /**
+     * Get the link for a specific tab on this ModelAdmin.
+     *
+     * If you do not have multiple tabs for a given class you should use {@link getLinkForModelClass}
+     * instead.
+     *
+     * @param string $modelTab This is either the custom url segment for the tab or, if that
+     * was not not defined, it is the fully qualified class name of the managed model.
+     * In the latter case, you should prefer to use {@link getLinkForModelClass}.
+     *
+     * @throws InvalidArgumentException if $modelTab is not a tab on this ModelAdmin.
+     */
+    public function getLinkForModelTab(string $modelTab): string
+    {
+        // Don't use isManagedModel here because a subclass of a managed model may not have its own tab.
+        if (!array_key_exists($modelTab, $this->getManagedModels())) {
+            throw new InvalidArgumentException("$modelTab isn't a tab on this ModelAdmin.");
+        }
+        return $this->Link($this->sanitiseClassName($modelTab));
+    }
+
+    /**
+     * Get the link for editing an object inside this ModelAdmin.
+     *
+     * @throws InvalidArgumentException if $obj is not managed by this ModelAdmin.
+     */
+    public function getEditLinkForDataObject(DataObject $obj): string
```

Comment:
Just wondering, would it make sense to change the signature in getEditLinkForDataObject to the following, and get rid of the extension:

```suggestion
    public function getEditLinkForDataObject(DataObject $obj, string $modelTab = ''): string
```

That could be another way to resolve the issue of the multiple tabs managing the same dataobject.  I think it would be more intuitive to put in the url-slug on the model tab than manually updating the edit link in via an extension class

Reply:
I think that boils down to whether the admin owns the concern or the object owns the concern.
From my point of view it should be the admin owning the concern - everything else about how the admin handles putting or pulls objects into/from different tabs is defined in the admin itself.
If the admin has multiple tabs with different filters, it should be the admin which says "The link for that object is in tab A, _not_ tab B". The object shouldn't care IMO